django/db/backends/oracle/operations.py

```diff
@@ -334,8 +336,20 @@ def regex_lookup(self, lookup_type):
             match_option = "'i'"
         return 'REGEXP_LIKE(%%s, %%s, %s)' % match_option
 
-    def return_insert_id(self, field):
-        return 'RETURNING %s INTO %%s', (InsertVar(field),)
+    def return_insert_columns(self, fields):
+        if not fields:
+            return '', ()
+        columns = []
+        params = ()
+        into_params = []
+        for field in fields:
+            columns.append("%s.%s" % (self.quote_name(field.model._meta.db_table), self.quote_name(field.column)))
+            params += (InsertVar(field),)
+            into_params.append("%s")
+        return "RETURNING %s INTO %s" % (
+            ", ".join(columns),
+            ", ".join(into_params)
```

Comment:
This could be `', '.join('%s' for i in len(columns)),` to avoid needing to create `into_params` up front.

Reply:
`TypeError: 'int' object is not iterable` ;) That could be fixed, but it is very consistent avoids calculating the list size.

Reply:
Sorry, I meant `range()`, not `len()`. (Or just ditch the `len()` and iterate over the elements.)